packages/wallet/src/services/util/equals.ts

```diff
@@ -11,7 +12,7 @@ export const arrayEquals = <T>(arrayA: T[], arrayB: T[], itemEquals: (a: T, b: T
 
 export const shallowArrayEquals = <T>(a: T[], b: T[]) => arrayEquals(a, b, strictEquals);
 
-export const tipEquals = (a: Cardano.Tip, b: Cardano.Tip) => a.hash === b.hash;
+export const tipEquals = (a: WC.Tip, b: WC.Tip) => a.slot === b.slot;
```

Comment:
This change might be a bug if there's a rollback to the same slot but different block